spring/spring-data/src/test/java/io/quarkus/ts/spring/data/rest/CrudRepositoryRestResourceIT.java

```diff
@@ -1,79 +1,65 @@
 
 import io.quarkus.test.scenarios.QuarkusScenario;
-import io.quarkus.ts.spring.data.AbstractDbIT;
 
 @QuarkusScenario
-public class CrudRepositoryRestResourceIT extends AbstractDbIT {
-    @Test
-    void testAllRepositoryMethods() throws InterruptedException {
+public class CrudRepositoryRestResourceIT extends AbstractRepositoryRestResourceIT {
 
-        //GET - List all libraries
-        app.given()
-                .accept("application/json")
-                .when().get("/library")
-                .then()
-                .statusCode(200)
-                .body(
-                        containsString("Library1"));
+    protected static final List<String> ORIGINAL_ITEMS = List.of("Library1");
+    protected static final String CREATED_ITEM = "Library2";
+    protected static final String UPDATED_ITEM = "Library Two";
 
-        //POST - Create a new Library
-        app.given()
-                .contentType("application/json")
-                .accept("application/json")
-                .body("{\"name\": \"Library2\"}")
-                .when().post("/library")
-                .then()
-                .statusCode(201)
-                .body(containsString("Library2"))
-                .body("id", notNullValue())
-                .extract().body().jsonPath().getString("id");
+    @Override
+    protected String getUrl() {
+        return "/library";
+    }
 
-        //GET{id} - Find new library by id
-        app.given()
-                .when().get("/library/id/2")
-                .then()
-                .statusCode(200)
-                .body(
-                        containsString("Library2"));
+    @Override
+    protected List<String> getOriginalItems() {
+        return ORIGINAL_ITEMS;
+    }
 
-        //PUT - Update library entry
-        app.given()
-                .contentType("application/json")
-                .accept("application/json")
-                .body("{\"name\": \"Library Two\"}")
-                .when().put("/library/2")
-                .then()
-                .statusCode(204);
+    @Override
+    protected String getCreatedItem() {
+        return CREATED_ITEM;
+    }
 
-        //GET{id} - Verify update
-        app.given()
-                .when().get("/library/id/2")
-                .then()
-                .statusCode(200)
-                .body(
-                        containsString("Library Two"));
+    @Override
+    protected long getCreatedItemId() {
+        return 2;
+    }
 
-        //DELETE - Delete a library
-        app.given()
-                .when().delete("/library/2")
-                .then()
-                .statusCode(204);
+    @Override
+    protected String getUpdatedItem() {
+        return UPDATED_ITEM;
+    }
+
+    @Override
+    protected String getItemNameHalJsonPath() {
+        return "_embedded.library.name";
+    }
+
+    @Override
+    protected String getItemIdUrl(long id) {
+        return getUrl() + "/id/" + id;
     }
 
+    @Order(12)
```

Comment:
A test with this number already exists[1]. And maybe it is not the best idea, to have global test order over several classes, but it is up to you.

[1] https://github.com/quarkus-qe/quarkus-test-suite/pull/853/files#diff-6e1ff9ae19b3166f50f5c958535c0f94279429bdc8b5bbb8473112c53fe63502R48

Reply:
That was intentional, they were in different descendant classes - two different implementation of the same test, because the entities and their constraints differ.

I've tried to minimize the number of tests with `@Order`, so it is no longer present in this particular method, but the 2 different implementations remained.